synfig-studio/src/gui/app.cpp

```diff
@@ -1673,67 +1677,13 @@ App::App(const synfig::String& basepath, int *argc, char ***argv):
 			splash_screen.show();
 		}
 
-		// Look for any files given on the command line,
-		// and load them if found.
-		for(;*argc>=1;(*argc)--)
-			if((*argv)[*argc] && (*argv)[*argc][0]!='-')
-			{
-				studio_init_cb.task(_("Loading files..."));
-				splash_screen.hide();
-				open((*argv)[*argc]);
-				opened_any = true;
-				splash_screen.show();
-			}
-
-		// if no file was specified to be opened, create a new document to help new users get started more easily
-		if (!opened_any && !getenv("SYNFIG_DISABLE_AUTOMATIC_DOCUMENT_CREATION"))
-			new_instance();
-
 		studio_init_cb.task(_("Done."));
 		studio_init_cb.amount_complete(10000,10000);
 
-		// To avoid problems with some window managers and gtk >= 2.18
-		// we should show dock dialogs after the settings load.
-		// If dock dialogs are shown before the settings are loaded,
-		// the windows manager can act over it.
-		// See discussions here:
-		// * https://synfig.org/forums/viewtopic.php?f=1&t=1131&st=0&sk=t&sd=a&start=30
-		// * https://synfig.org/forums/viewtopic.php?f=15&t=1062
-		dock_manager->show_all_dock_dialogs();
-
-		main_window->present();
-
```

Comment:
Are you sure you can safely remove it?

Reply:
I mistakenly removed the ```dock_manager->show_all_dock_dialogs();``` ! Will be added in next commit :)

Reply:
My bad. I mistakenly removed ``` dock_manager->show_all_dock_dialogs(); ``` ! Will be added in next commit for sure :) Thanks !

Reply:
It's important to not delete anything while you are 'studying' for not losing any expected behavior that is already implemented.
You could comment-out the code you are unsure if it could be removed, until you are sure. Look at original Line 1689, for example.

Reply:
Ok ! Thanks for telling me ! I will comment the code first from now on ! I thought line 1689 won't get used after implementing Gtk::Application because on_open() and on_activate() are seperated now ! and on_open() will only be called if there are files to open ! Kindly correct me if I am wrong :)

Thanks again ! I will comment first from now onwards ! Readding the code I removed mistakenly :)